sqld/src/replication/primary/logger.rs

```diff
@@ -616,7 +659,28 @@ mod test {
             data: vec![0; 3].into(),
         };
 
-        let (count, chk) = logger.write_pages(&[entry]).unwrap();
-        logger.commit(count, chk);
+        logger.write_pages(&[entry]).unwrap();
+        logger.commit().unwrap();
+    }
+
+    #[test]
+    fn log_file_test_rollback() {
```

Comment:
This test checks if we're only able to read committed entries, ignoring the uncommitted ones, right? I suggest extending it a little to match the name, and actually add a rollback() step:
1. Write a few frames
2. Verify they aren't visible
3. Commit
4. Write some more frames
5. Rollback
6. Verify that only committed frames are there
7. Write one more frame
8. Commit
9. Verify that no garbage from rolled-back frames is visible after we overwritten one of them in the wallog file